core/lib/multivm/src/versions/vm_latest/utils/v26_upgrade.rs

```diff
@@ -0,0 +1,132 @@
+//! This module contains varios v26-upgrade-related functions to be used by the state keeper's
+//! unit tests.
+//! The correctness of the functionality present here is hard to enforce inside state keeper
+//! directly, so it is done inside unit tests of the multivm.
+
+use std::{
+    collections::HashMap,
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+
+use ethabi::Token;
+use zksync_contracts::{l2_asset_router, l2_legacy_shared_bridge};
+use zksync_types::{utils::encode_ntv_asset_id, Address, StorageKey, H256, U256};
+use zksync_utils::env::Workspace;
+
+#[derive(Debug, PartialEq, Eq)]
+pub struct V26TestData {
+    pub l1_chain_id: U256,
+    pub l1_shared_bridge_address: Address,
+    pub l1_token_address: Address,
+    pub l2_token_address: Address,
+    pub l2_legacy_shared_bridge_address: Address,
+    pub l1_aliased_shared_bridge: Address,
+}
+
+pub fn get_test_data() -> V26TestData {
+    V26TestData {
+        l1_chain_id: 1.into(),
+        l1_shared_bridge_address: Address::from_str("abacabac00000000000000000000000000000001")
+            .unwrap(),
+        l1_token_address: Address::from_str("abacabac00000000000000000000000000000002").unwrap(),
+        l2_token_address: Address::from_str("de1b93723f716c741b771276465b7b58aa9c44a6").unwrap(),
+        l2_legacy_shared_bridge_address: Address::from_str(
+            "d316600b6009f94ab776e97f1fc985bc18b4e535",
+        )
+        .unwrap(),
+        l1_aliased_shared_bridge: Address::from_str("bcbdabac00000000000000000000000000001112")
+            .unwrap(),
+    }
+}
+
+fn get_test_v26_path(name: &str) -> PathBuf {
```

Comment:
Assuming retaining JSON files is necessary (see the other line comment; after reduction, the *may* be replaceable by inline definitions): The approach with locating a workspace looks fragile / unnecessarily complex. A simpler approach would be to use `include_str!` and copy JSON files to the state keeper crate via symlinks. In this case, there would be some duplication deserializing files, but it should be fairly short. Further, in this case JSON files could be encapsulated in unit tests for each crate (i.e., not compiled as a part of the `multivm` crate API).